syntax = "proto3";

package user;

service UserService {
    rpc CreateUser(CreateUserRequest) returns (UserResponse);
    rpc GetUser(GetUserRequest) returns (UserResponse);
    rpc AuthenticateUser(AuthRequest) returns (AuthResponse);
}

message CreateUserRequest {
    string username = 1;
    string email = 2;
    string password = 3;
}

message UserResponse {
    string user_id = 1;
    string username = 2;
    string email = 3;
}

message GetUserRequest {
    string user_id = 1;
}

message AuthRequest {
    string username = 1;
    string password = 2;
}

message AuthResponse {
    bool success = 1;
    string token = 2;
    string error = 3;
}
